lib/matplotlib/tests/test_figure.py

```diff
@@ -688,6 +688,86 @@ def test_removed_axis():
     fig.canvas.draw()
 
 
+def test_figure_clear():
```

Comment:
I have a mild preference to split each of these cases into its own test, but maybe that you can keep re-using the same figure is part of the test?